presto-main/src/test/java/io/prestosql/tests/ReportUnannotatedMethods.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests;
+
+import com.google.common.reflect.TypeToken;
+import org.testng.IClassListener;
+import org.testng.ITestClass;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
+
+public class ReportUnannotatedMethods
+        implements IClassListener
+{
+    @Override
+    public void onBeforeClass(ITestClass testClass)
+    {
+        Method[] publicMethods = testClass.getRealClass().getMethods();
+
+        List<Method> unannotatedMethods = Arrays.stream(publicMethods)
+                .filter(method -> method.getDeclaringClass() != Object.class)
+                .filter(method -> !Modifier.isStatic(method.getModifiers()))
+                .filter(method -> !method.isBridge())
+                .filter(method -> !isTestMethod(method))
+                .collect(toImmutableList());
+
+        if (!unannotatedMethods.isEmpty()) {
+            throw new IllegalStateException(format(
```

Comment:
`ISE` doesn't seem to fit here. Should probably be `AssertionError` or `RuntimeException`

Reply:
AE looks nice, but i didn't want to throw Error, not to trigger some weird behavior in testng/surefire/somewhere. I will stick to RTE.